### PR TITLE
fix(ReportDetails): RHICOMPL-1138 Show hosts with policy results

### PR DESCRIPTION
**Pull request by akofink**
This changes the systems table to accept a defaultFilter prop which will
be added to user filters during the API request. This is to allow
specifying a new filter for the ReportDetails page to only show hosts
that have test results for a given policy (a set of profiles associated
to the policy). The new filter takes policy ID or any profile ID
associated to the policy. It also reports test result hosts for external policy profiles.

Requires https://github.com/RedHatInsights/compliance-backend/pull/678

Signed-off-by: Andrew Kofink <akofink@redhat.com>

### CODE REVIEW

File: src/SmartComponents/SystemsTable/SystemsTable.js [hunk 1]
```diff
@@ -176,19 +176,15 @@ class SystemsTable extends React.Component {
...
+            this.filterBuilder.buildFilterString(activeFilters)
+        ].filter((e)=>!!e).join(' and ');
 
-        if (showOnlySystemsWithTestResults) {
-            filter = `has_test_results = true ${filter.length > 0 ? `and ${filter}` : ''}`;
-        }
-
-        let variables = { filter, perPage, page };
-        if (policyId) {
-            filter = `policy_id = ${policyId} and ${filter}`;
```
Comment (bastilian) [inline src/SmartComponents/SystemsTable/SystemsTable.js:189]: Do we not want need to use this filter anymore?

Comment (vkrizan) [reply src/SmartComponents/SystemsTable/SystemsTable.js:189]: Indeed. This affects showing assigned-only systems on a SCAP Policy. Mind to adding a default filter for that case (`PolicyDetails`). Also please check other usages, like policy edit.

**Review by vkrizan — CHANGES_REQUESTED** [review]

**Comment by vkrizan** [comment]
Works correctly on the reports page. Thanks.

File: src/SmartComponents/SystemsTable/SystemsTable.js (hunk 1 above)
Comment (akofink) [reply src/SmartComponents/SystemsTable/SystemsTable.js:189]: Whoops! Sorry, I meant to check for this and forgot :see_no_evil: Updated!

**Review by vkrizan — APPROVED** [review]
Seems to work on all pages. Thank you.

**Pull request merged**